src/metaopt/io/database/mongodb.py

```diff
@@ -86,10 +86,10 @@ def unlock(self):
             raise_from(DatabaseError("Could not determine unlock due to connection."), e)
         return unlock
 
-    def insert(self, collection_name, query, dictionaries):
+    def insert(self, collection_name, query, document_dictionaries):
```

Comment:
What about the query? Based on the [doc](http://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.insert_many), it seems it should be `insert_many(document_dictionaries)`. I don't understand what a query would be used for in during an insertion.